File: project/app/src/main/java/org/owntracks/android/ui/regions/RegionsAdapter.java
package org.owntracks.android.ui.regions;

import androidx.annotation.NonNull;
import android.view.View;

import org.owntracks.android.BR;
import org.owntracks.android.R;
import org.owntracks.android.data.WaypointModel;
import org.owntracks.android.ui.base.BaseAdapter;
import org.owntracks.android.ui.base.BaseAdapterItemView;

import java.util.List;

import io.objectbox.reactive.DataObserver;


class RegionsAdapter extends BaseAdapter<WaypointModel> implements DataObserver<List<WaypointModel>> {
    RegionsAdapter(ClickListener clickListener) {
        super(BaseAdapterItemView.of(BR.waypoint, R.layout.ui_row_region));
        setClickListener(clickListener);
    }

    @Override
    public void onData(@NonNull List<WaypointModel> data) {
        setItems(data);
    }

    interface ClickListener extends BaseAdapter.ClickListener<WaypointModel> {
        void onClick(@NonNull WaypointModel object, @NonNull View view, boolean longClick);
    }
}

File: project/app/src/main/java/org/owntracks/android/ui/regions/RegionsAdapter.kt
package org.owntracks.android.ui.regions

import androidx.databinding.ViewDataBinding
import org.owntracks.android.BR
import org.owntracks.android.R
import org.owntracks.android.data.WaypointModel
import org.owntracks.android.ui.base.BaseRecyclerViewAdapterWithClickHandler
import org.owntracks.android.ui.base.BaseRecyclerViewHolder

class RegionsAdapter(clickListener: ClickListener<WaypointModel>) :
    BaseRecyclerViewAdapterWithClickHandler<WaypointModel, RegionsAdapter.WaypointModelViewHolder>(
        clickListener, ::WaypointModelViewHolder, R.layout.ui_row_region
    ) {

    class WaypointModelViewHolder(binding: ViewDataBinding) :
        BaseRecyclerViewHolder<WaypointModel>(binding, BR.waypoint)
}
